### PR TITLE
[MAINT] remove skip doctest items for old numpy

### LINKED ISSUE #3611: remove `pytest_collection_modifyitems` from conftest.py
Do we really need this ?
I think we are only supporting 1.19 and above these days.

_Originally posted by @NicolasGensollen in https://github.com/nilearn/nilearn/pull/3609#discussion_r1145977539_

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3611 

Changes proposed in this pull request:

- [remove skip doctest items for old numpy](https://github.com/nilearn/nilearn/commit/f597f2c72d649b421953bebc8acb3cb94698f89e)

### CODE REVIEW

File: nilearn/conftest.py [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
 import numpy as np
 import pytest
-from _pytest.doctest import DoctestItem
 from nilearn import image
 
 # we need to import these fixtures even if not used in this module
 from nilearn.datasets._testing import request_mocker  # noqa: F401
 from nilearn.datasets._testing import temp_nilearn_data_dir  # noqa: F401
-from nilearn.version import _compare_version
+from nilearn.version import _compare_version  # noqa: F401
```
Comment (NicolasGensollen) [inline nilearn/conftest.py:10]: Why didn't you remove the import ?

**Review by NicolasGensollen — APPROVED** [review]
Thanks for taking care of this @Remi-Gau !
LGTM, I just have one question.

File: nilearn/conftest.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/conftest.py]: I think I got confused by the comment above that those imports were needed.
Removed it.
Let's see what CI says.

**Review by bthirion — APPROVED** [review]
LGTM, hx.

**Pull request merged**